src/sensai/data_transformation.py

```diff
@@ -39,44 +100,73 @@ def applyInverse(self, df: pd.DataFrame) -> pd.DataFrame:
 class RuleBasedDataFrameTransformer(DataFrameTransformer, ABC):
     """Base class for transformers whose logic is entirely based on rules and does not need to be fitted to data"""
 
+    def _fit(self, df: pd.DataFrame):
+        pass
+
     def fit(self, df: pd.DataFrame):
         pass
 
+    def isFitted(self):
+        return True
 
-class DataFrameTransformerChain:
-    """Supports the application of a chain of data frame transformers"""
 
-    def __init__(self, dataFrameTransformers: Sequence[DataFrameTransformer]):
-        self.dataFrameTransformers = dataFrameTransformers
+class DataFrameTransformerChain(DataFrameTransformer):
+    """
+    Supports the application of a chain of data frame transformers.
+    During fit and apply each transformer in the chain receives the transformed output of its predecessor.
+    """
+    @staticmethod
+    def _flattened(l: Sequence[Union[DataFrameTransformer, Sequence[DataFrameTransformer]]]) -> List[DataFrameTransformer]:
+        result = []
+        for x in l:
+            if isinstance(x, DataFrameTransformer):
+                result.append(x)
+            else:
+                result.extend(x)
+        return result
 
-    def apply(self, df: pd.DataFrame, fit=False) -> pd.DataFrame:
-        """
-        Applies the chain of transformers to the given DataFrame, optionally fitting each transformer before applying it.
-        Each transformer in the chain receives the transformed output of its predecessor.
+    # TODO: do we really need the more flexible interface? I propose a breaking change to *dataFrameTransformers: DataFrameTransformer
```

Comment:
Why can't we just keep the old interface? What was the reason for the change in the first place?

Reply:
Just consistency/simplicity (I am always thinking about the "released stable API"). Since this is addressed in other comments as well, I am resolving this conversation.

Reply:
Consistency with what?
Simplicity: no, the old interface is much simpler.

Reply:
Consistency with VectorModel here. The with... methods used this more relaxed interface while the chain had a different one. Anyway, I will use the to-be-created util.flatten method here (tomorrow).

Reply:
I understand your point and indeed it is uniformity, not consistency that I am after. Sorry for confusing the two, they are different notions indeed. It is true that uniformity is not necessarily an improvement and this particular change is mostly about my personal preference (I wanted to be able to use the same interface in Chained and Multi fgens). Since we agreed on the flattened solution and we have other open threads addressing it, I propose to close this thread